README.md

```diff
@@ -4,37 +4,680 @@ Sylvia is the old name meaning Spirit of The Wood.
+
+    #[msg(exec)]
+    fn add_member(&self, ctx: ExecCtx, member: String) -> Result<Response, ContractError> {
+        let member = ctx.deps.api.addr_validate(member)?;
+        self.members.save(ctx.deps.storage, &member, &Empty {})?;
+        Ok(Response::new())
+    }
+
+    #[msg(query)]
+    fn is_member(&self, ctx: QueryCtx, member: String) -> Result<group::IsMemberResp, ContractError> {
+        let is_member = self.members.has(ctx.deps.storage, &Addr::unchecked(&member));
+        let resp = group::IsMemberResp {
+            is_member,
+        };
+
+        Ok(resp)
+    }
+}
+
+#[contract]
+#[messages(group as Group)]
+impl MyContract<'_> {
+    // Nothing changed here
+}
+```
+
+Here are a couple of things to talk about.
+
+First, note that I defined the interface trait in its separate module with a name
+matching the trait name. This is a requirement right now - Sylvia generates all
+the messages and boilerplate in this module and will try to access them through
+this module.
+
+Then there is the `Error` type embedded in the trait - it is also needed there,
+and the trait bound here has to be at least `From<StdError>`, as Sylvia might
+generate code returning an `StdError` in deserialization/dispatching implementation.
+The trait can be more strict - this is the minimum.
+
+Another thing to remember is that the `#[msg(...)]` attributes become part of the
+function signature - they must be the same for the trait and later implementation.
+
+Finally, every implementation block has an additional
+`#[messages(module as Identifier)]` attribute. Sylvia needs it to generate the dispatching
+properly - there is the limitation that every macro has access only to its local
+item. In particular - we cannot see all traits implemented by a type and their
+implementation from the `#[contract]` crate.
+
+To solve this issue, we put this `#[messages(...)]` attribute pointing to Sylvia
+what is the module name where the interface is defined, and giving a unique name
+for this interface (it would be used in generated code to provide proper enum variant).
+
+The impl-block with trait implementation also contains the `#[messages]` attribute,
+but only one - the one with info about the trait being implemented.
+
+## Usage in external crates
+
+What is important is the possibility of using generated code in the external code.
+First, let's start with generating the documentation of the crate:
+
+```sh
+cargo doc --document-private-items --open
+```
+
+This generates and opens documentation of the crate, including all generated structures.
+`--document-private-item` is optional, but it will generate documentation of not-public
+modules which is sometimes useful.
+
+Going through the doc, you will see that all messages are generated in their structs/traits
+modules. To send messages to the contract, we can just use them:
+
+```rust
+use sylvia::cw_std::{WasmMsg, to_binary};
+
+fn some_handler(my_contract_addr: String) -> StdResult<Response> {
```

Comment:
Let's make this easier with #28 and you can update this code to show how to use the handlers...
Not blocking a merge, but a note for later

Reply:
#28 is not yet implemented. It Is README for the state of today. Will be updated before 0.5 release